mozci/push.py

```diff
@@ -298,7 +306,11 @@ def tasks(self):
                 "push_tasks_classifications", branch=self.branch, rev=self.rev
             )
             for task in tasks:
-                if task["id"] in classifications:
+                # No need to gather task classifications for a completed task that was already cached
+                if (
+                    task["id"] not in completed_cached_tasks
+                    and task["id"] in classifications
+                ):
```

Comment:
Classifications might actually change (sheriffs can classify a few hours after a failure), so let's keep the update no matter what (it's not expensive anyway, the expensive part is the `push_tasks_classifications` query above, which we have to do anyway).